std/typecons.d

```diff
@@ -759,31 +759,79 @@ template Tuple(Specs...)
         {
-            enum header = typeof(this).stringof ~ "(",
-                 footer = ")",
-                 separator = ", ";
-            sink(header);
-            foreach (i, Type; Types)
+            if (fmt.nested)
             {
-                static if (i > 0)
+                import std.format : formattedWrite;
+                if (fmt.sep)
                 {
-                    sink(separator);
+                    foreach (i, Type; Types)
+                    {
+                        static if (i > 0)
+                        {
+                            sink(fmt.sep);
+                        }
+                        // TODO: Change this once toString() works for shared objects.
+                        static if (is(Type == class) && is(Type == shared))
+                        {
+                            sink(Type.stringof);
+                        }
+                        else
+                        {
+                            formattedWrite(sink, fmt.nested, this.field[i]);
+                        }
+                    }
                 }
-                // TODO: Change this once toString() works for shared objects.
-                static if (is(Type == class) && is(typeof(Type.init) == shared))
+                else
                 {
-                    sink(Type.stringof);
+                    formattedWrite(sink, fmt.nested, this.expand);
                 }
-                else
+            }
+            else if (fmt.spec == 's')
+            {
+                enum header = Unqual!(typeof(this)).stringof ~ "(",
+                     footer = ")",
+                     separator = ", ";
+                sink(header);
+                foreach (i, Type; Types)
                 {
-                    import std.format : FormatSpec, formatElement;
-                    FormatSpec!char f;
-                    formatElement(sink, field[i], f);
+                    static if (i > 0)
+                    {
+                        sink(separator);
+                    }
+                    // TODO: Change this once toString() works for shared objects.
+                    static if (is(Type == class) && is(Type == shared))
+                    {
+                        sink(Type.stringof);
+                    }
+                    else
+                    {
+                        import std.format : FormatSpec, formatElement;
+                        // FormatSpec!Char
+                        FormatSpec!char f;
+                        formatElement(sink, field[i], f);
+                    }
                 }
+                sink(footer);
+            }
+            else
+            {
+                import std.format : FormatException;
+                throw new FormatException(
+                    "Expected '%s' or '%(...%)' or '%(...%|...%)' format specifier for type '" ~ Unqual!(typeof(this)).stringof ~ "', not '%" ~ [ fmt.spec ] ~ "'.");
```

Comment:
The brackets around `fmt.spec` shouldn't be needed; `char` should be able to be concatenated with slices of `char` freely. It is an unnecessary GC allocation.